src/schnorrsig.rs

```diff
@@ -289,18 +292,57 @@ impl PublicKey {
                 return Err(Error::InvalidTweak);
             }
 
+            let mut parity: ::secp256k1_sys::types::c_int = 0;
             err = ffi::secp256k1_xonly_pubkey_from_pubkey(
                 secp.ctx,
-                &mut self.0 as *mut _,
-                ptr::null_mut(),
+                &mut self.0,
+                &mut parity,
                 &pubkey,
             );
 
-            return if err == 0 {
+            if err == 0 {
                 Err(Error::InvalidPublicKey)
             } else {
+                Ok(parity != 0)
+            }
+        }
+    }
+
+    /// Verify that a tweak produced by `tweak_add_assign` was computed correctly
+    ///
+    /// Should be called on the original untweaked key. Takes the tweaked key and
+    /// output parity from `tweak_add_assign` as input.
+    ///
+    /// Currently this is not much more efficient than just recomputing the tweak
+    /// and checking equality. However, in future this API will support batch
+    /// verification, which is significantly faster, so it is wise to design
+    /// protocols with this in mind.
+    pub fn tweak_add_check<V: Verification>(
+        &self,
+        secp: &Secp256k1<V>,
+        tweaked_key: &Self,
+        tweaked_parity: bool,
+        tweak: &[u8],
+    ) -> Result<(), Error> {
```

Comment:
What do you think about passing `tweak: [u8; 32]` and then returning `bool` instead of a Result?
In general I feel like it would be nicer to use the type system to require sizes instead of adding dozens of results/errors (especially with `TryFrom` that let's you `slice.try_into()?` for converting to array, but that's 1.34)

Reply:
if for ergonomic pre 1.34 reasons you want to keep it as a slice then that's ok

Reply:
Yeah I think it'd be reasonable to use arrays here.

Reply:
Welll, for what it's worth we already have an error return here if the user passes an invalid tweak. If you know you're giving a hash then it can't fail, but that's still an error you have to `.unwrap` away. 

I'm not sure there's any additonal ergonomics loss to using a slice.